jpa/eclipselink.jpa.testapps/jpa.test.persistence32/src/test/java/org/eclipse/persistence/testing/tests/jpa/persistence32/QueryTest.java

```diff
@@ -11,10 +12,15 @@
  */
 package org.eclipse.persistence.testing.tests.jpa.persistence32;
 
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
```

Comment:
There is no single use of hamcrest in the repo and the project itself is happy without it, so do not add it, please.

Reply:
Removed Hamcrest assertions. The hamcrest dependency is transitively brought to the project by junit. Maybe it would be good to exclude hamcrest from the project. I wouldn't use it if I saw it's excluded.

Reply:
inability to avoid hamcrest dependency is known junit 3/4 issue - see ie https://github.com/junit-team/junit4/issues/1145 and/or https://github.com/junit-team/junit4/issues/1741 if you're interested in details

Reply:
This is no longer true with latest versions of JUnit 4. The Hamcrest dependency of JUnit can be excluded, see: https://github.com/OmniFish-EE/eclipselink/commit/1e6a11c47cf762950ac064363ad89c6a0666e49f. Should I raise a PR to do it?

Reply:
> This is no longer true with latest versions of JUnit 4. The Hamcrest dependency of JUnit can be excluded, see: [OmniFish-EE@1e6a11c](https://github.com/OmniFish-EE/eclipselink/commit/1e6a11c47cf762950ac064363ad89c6a0666e49f). Should I raise a PR to do it?

I did it in https://github.com/eclipse-ee4j/eclipselink/pull/2226

Reply:
> This is no longer true with latest versions of JUnit 4. The Hamcrest dependency of JUnit can be excluded, see: [OmniFish-EE@1e6a11c](https://github.com/OmniFish-EE/eclipselink/commit/1e6a11c47cf762950ac064363ad89c6a0666e49f). Should I raise a PR to do it?

It looks not so easy in EclipseLink project. There are test failures in #2226 
`mvn clean verify -pl :org.eclipse.persistence.jpa.jpql` throws exception in `org.eclipse.persistence.jpa.tests.jpql.AllHermesTests` . I think it's releated with `@RunWith`, `org.junit.runner.RunWith`.

Reply:
https://github.com/eclipse-ee4j/eclipselink/actions/runs/10266202341/job/28404036811

Reply:
It looks like hamcrest is used somewhere. Maybe by JUnit itself. Here seems to be the same problem: https://github.com/eclipse-ee4j/eclipselink/pull/261. At that time, it looks like it was necessary to add hamcrest on the classpath in Ant configuration. But the problem was the same - something needed Hamcrest.

Reply:
Yes, JUnit 4 uses hamcrest in several places: https://github.com/search?q=repo%3Ajunit-team%2Fjunit4+hamcrest+language%3AJava&type=code&l=Java. So it's not possible to completely remove it.

Reply:
I think it's better to let it be. Or introduce a checkstyle rule that fails if hamcrest packages are imported.

Reply:
A solution could be to make hamcrest a runtime dependency. JUnit would work but it couldn't be used in compiled code.